### PR TITLE
Bump redis to 4.4.0

### PR DESCRIPTION
**Pull request by juanamari94**
Closes #9329

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Thanks! You need to update this value here, otherwise the CI will compare the stubs you're writing to the 4.3.5 branch at runtime, which will lead to the CI failing :)

https://github.com/python/typeshed/blob/main/stubs/redis/METADATA.toml#L1

**Comment by juanamari94** [comment]
Hey @AlexWaygood ! Appreciate the support while I get started with the contribution. Will update it!

**Comment by juanamari94** [comment]
Hey @AlexWaygood ! I think I'm done with most changes, I'm just coming across an error that I'm not certain how to tackle:

```
error: redis.asyncio.sentinel.SentinelManagedConnection.read_response is inconsistent, runtime does not have argument "timeout"
Stub: at line 18 in file /home/runner/work/typeshed/typeshed/stubs/redis/redis/asyncio/sentinel.pyi
def (self: redis.asyncio.sentinel.SentinelManagedConnection, disable_decoding: builtins.bool =, timeout: Union[builtins.float, None] =) -> typing.Coroutine[Any, Any, Any]
Runtime: at line 66 in file /tmp/tmpm_t04cur/lib/python3.10/site-packages/redis/asyncio/sentinel.py
async def (self, disable_decoding: bool = False)
```

It's complaining about the argument `timeout` being present where it shouldn't be, but if I remove it then the stub tests will require for it to be present.

[I've noticed that this change was recently merged into redis-py last week](https://github.com/redis/redis-py/blame/master/redis/asyncio/sentinel.py#L66), could this help pinpoint the underlying issue?

File: stubs/redis/redis/asyncio/sentinel.pyi [hunk 1]
```diff
@@ -15,7 +15,7 @@ class SentinelManagedConnection(Connection):
     def __init__(self, **kwargs) -> None: ...
     async def connect_to(self, address) -> None: ...
     async def connect(self): ...
-    async def read_response(self, disable_decoding: bool = ...): ...
+    async def read_response(self, disable_decoding: bool = ..., timeout: Optional[float] = ...): ...
```
Comment (AlexWaygood) [inline stubs/redis/redis/asyncio/sentinel.pyi:18]: This should fix it :)

```suggestion
    async def read_response(self, disable_decoding: bool = ...): ...  # type: ignore[override]
```

**Comment by AlexWaygood** [comment]
Thanks @juanamari94! Could you possibly take a look at the `check_new_syntax.py` linting failures?

**Comment by juanamari94** [comment]
@AlexWaygood done! Are you in favor of rebases in this codebase? I'd like to rebase the commits of this branch into a single one to keep the history tidy.

**Comment by AlexWaygood** [comment]
> @AlexWaygood done! Are you in favor of rebases in this codebase? I'd like to rebase the commits of this branch into a single one to keep the history tidy.

In general we prefer not to rebase wherever possible, as it interacts badly with GitHub's UI, and makes incremental reviews difficult :)

It'll all be squash-merged into one commit at the end anyway, so you don't need to worry about making a mess!

**Comment by juanamari94** [comment]
@AlexWaygood Gotcha! Let me know if you'd like to see any amended commits either way

**Comment by juanamari94** [comment]
@AlexWaygood Appreciate the `CredentialProvider` fix!

**Comment by AlexWaygood** [comment]
> @AlexWaygood Appreciate the `CredentialProvider` fix!

No problem! Hope you don't mind me pushing to your branch, it felt easier here :)

**Review by AlexWaygood — APPROVED** [review]
Thanks, this is really helpful! Will merge once the CI's finished 😀

**Comment by juanamari94** [comment]
@AlexWaygood That's alright! Appreciate all the help and the very prompt replies 🙏

**Pull request merged**